tracer/src/Datadog.Trace/Util/AsyncUtil.cs

```diff
@@ -77,4 +109,39 @@ public static TResult RunSync<TResult>(Func<Task<TResult>> task, CancellationTok
           .Unwrap()
           .GetAwaiter()
           .GetResult();
+
+    /// <summary>
+    /// Executes an async Task[T] method which has a T return value synchronously
+    /// USAGE: AsyncUtil.RunSync(() => AsyncMethod[T](), millisecondsTimeout);
+    /// </summary>
+    /// <typeparam name="TResult">Return Type</typeparam>
+    /// <param name="task">Task[T] method to execute</param>
+    /// <param name="millisecondsTimeout">Timeout in milliseconds</param>
+    public static TResult RunSync<TResult>(Func<Task<TResult>> task, int millisecondsTimeout)
+    {
+        return _taskFactory
+              .StartNew(TaskWithTimeoutAsync)
+              .Unwrap()
+              .GetAwaiter()
+              .GetResult();
+
+        Task<TResult> TaskWithTimeoutAsync()
+        {
+            var runTask = task();
+            return runTask.IsCompleted ? runTask : InternalTaskWithTimeoutAsync(runTask, millisecondsTimeout);
```

Comment:
built-in in .NET 6 (same as above)

Reply:
done in https://github.com/DataDog/dd-trace-dotnet/pull/3609/commits/18bfc35e3ce644ce86d4f155a7ffde9667c3a1d8